File: client/src/app/models/Content.ts
import { Account } from "./Account";

export interface Content {
    id: string;
    created: string | null;
    message: string;
    accountId: number;
    account: Account;
    contentImages: ContentImage[];
}

export interface ContentImage {
    id: number;
    imageUrl: string;
    contentId: string;
}

export interface ContentRequest {  
    Message : any;
    AccountId : any;
    FormFiles : any;
}

File: api/api/Models/Content.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace api.Models
{
    public class Content
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.None)]
        public string Id { get; set; }
        public DateTime? Created { get; set; }
        public string Message { get; set; } = string.Empty;
        public int AccountId { get; set; }
        public Account Account { get; set; }
        public List<ContentImage> ContentImages { get; set; } = new();
        public List<Comment> Comments { get; set; }
    }
}
